ddtrace/contrib/aiohttp/patch.py

```diff
@@ -88,7 +88,7 @@ async def _traced_clientsession_request(aiohttp, pin, func, instance, args, kwar
             span.service = url.host
 
         if pin._config["distributed_tracing"]:
-            HTTPPropagator.inject(span.context, headers)
+            HTTPPropagator.inject(span.context, headers, span)
```

Comment:
I doubt many users are calling HTTPPropagator.inject to begin with, but if they are, without passing in the span, this line will almost certainly grab the correct span regardless: https://github.com/DataDog/dd-trace-py/pull/8308/files#diff-db5c2c06582af0a924e93e08a60bea035ff4de9bf97a08bc12b7098c3546c8dcR958-R960

Noting other option as well:
We could just completely remove this kwarg and rely on the above referenced code to always grab us the root span. When I spoke with @emmettbutler  about this he thought it would be better if we could be explicit when it was an option, and therefore I stuck to that. 
I don't know in what circumstances `tracer.current_root_span` would fail us. I think I'm fine with either.
```
 def current_root_span(self) -> Optional[Span]:
        span = self.current_span()
        if span is None:
            return None
        return span._local_root
```

Reply:
Spoke with Kyle, just rely on tracer.current_root_span

Reply:
Makes sense. I hadn't really thought about the functional "breaking change" incurred by expecting a span argument without a default. The real issue is that `current_root_span` must be the default whether or not there's the option to pass in a span.

Reply:
Also down the line, would be good to have inject be a tracer method, since it already has all of the info.